### ISSUE TITLE
Inconsistent funnel collection rate for vehicle/placed.

### ISSUE THREAD

**Issue by Beanzy** [CleverRaven/Cataclysm-DDA#14063]
OS: Windows 7 x64
Cata Version: Tiles x64, Build 3910
Issue: Vehicle funnels appear to have a different collection rate than funnels placed on the ground.

I thought something seemed off, so I debugged the weather to rainy and placed a funnel on the ground and one on a vehicle; the ground funnel collected ~6x more water than the vehicle one did in the same amount of time.

**Comment by Beanzy** [comment]
I thought so too, but if that's the case, then I think the item description should probably indicate that a vehicle mounted funnel will be less efficient.

**Comment by Oskar636** [comment]
In build 5864 windows 64bit tiles, it is now so that car funnels collect twice the amount of water as the ground ones.

**Comment by Leland** [comment]
Still around. From #21094

>Put a funnel on a 60l tank sitting on the ground, and installed a funnel over a empty 60l tank on a vehicle. In the same amount of time the tank on the ground accumulated 2.5 liters the vehicle only accumulated 1 liter

**Comment by 1skandar** [comment]
Yeah, if I'm reading the code correctly, no guarantees I am, funnels have a charges per turn they collect, modified by the weather. Funnels on the ground appear to use this straight, vehicle mounted funnels appear to adjust this by the funnel's volume. If this behavior isn't intended that should be an easy fix. If I'm reading things correctly.

**Comment by Toothspit** [comment]
I can't repro this, and I've been going over this (and over, and over) for like an hour after chasing down where I thought the problem should live and spending far too long in gdb.

When testing, keep in mind that debug'd weathers *will not* properly fill vehicle funnels, as those [use](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/vehicle.cpp#L5967) the [weather generator and game seed](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/weather.cpp#L134) to [reconstruct](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/weather.cpp#L86) the rain that 'should have' fallen over the past [minute](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/vehicle.cpp#L5953). This behavior [is shared](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/weather.cpp#L152) with funnels that need to be back-filled because they were out of the reality bubble, and is presumably done for performance concerns.

To confirm that this cannot be repro'd, [here's](https://github.com/CleverRaven/Cataclysm-DDA/files/1027669/issue14063.zip) a save. Wait through a couple of weather cycles until a few liters of rain collects in the ground tank to your immediate right, you'll have roughly the same levels in the vehicle tanks. [Here's a video of me doing that for you](https://asciinema.org/a/bss3amxwbq5clfdgkbnauzwfl), multiply the L in vehicle tanks by 4 to get 'units' on the ground, all tanks have about 50 units. There is an rng to it, but the very same function is used for ground funnels, `roll_remainder` (to truncate floats to ints a little better).

**Comment by 1skandar** [comment]
Odd, I'll have to double check this. In my case it was raining when the game started. I had to debug the vehicle and parts in, but the rain wasnt. 3 or so hours of rainfall produced roughly 2 to 3 times the water for the tank on the ground as the one in the vehicle.

**Comment by Toothspit** [comment]
Can you post a save? I tried my best, but I'm willing to be wrong here!

**Comment by 1skandar** [comment]
Yes. Might be a few hours, though, as I'm a bit busy. *Something* is certainly up, though, and the code for funnels on the ground is decidedly different than vehicle funnels, regardless.

**Comment by 1skandar** [comment]
https://dl.dropboxusercontent.com/u/46780505/Pennsauken.zip

Ok, just retested. Not as bad this time, but still different. Funnel on ground: 10.5 liters. Funnel on vehicle, 8.5 liters. 

I've got to try a summer start, next. Something is decidedly up.

**Comment by Maleclypse** [comment]
Is the vehicle moving when during the test time frame?
